model/api/src/main/java/org/jboss/aerogear/unifiedpush/api/WebPushRegistration.java

```diff
@@ -0,0 +1,85 @@
+package org.jboss.aerogear.unifiedpush.api;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.Base64;
+
+/**
+ * This is a deserialized webpush token.
+ */
+public class WebPushRegistration {
+    private String endpoint ="";
+    private PushKey keys = new PushKey();
+
+    @JsonIgnore
+    public byte[] getAuthAsBytes() {
+        return Base64.getDecoder().decode(keys.getAuth());
+    }
+
+    @JsonIgnore
+    public byte[] getKeyAsBytes() {
+        return Base64.getDecoder().decode(keys.getP256dh());
+    }
+
+    /**
+     * @param endpoint the endpoint to set
+     */
+    public void setEndpoint(String endpoint) {
+        this.endpoint = endpoint;
+    }
+
+    /**
+     * @param keys the keys to set
+     */
+    public void setKeys(PushKey keys) {
+        this.keys = keys;
+    }
+
+    /**
+     * @return the endpoint
+     */
+    public String getEndpoint() {
+        return endpoint;
+    }
+    /**
+     * @return the keys
+     */
+    public PushKey getKeys() {
+        return keys;
```

Comment:
This will return a mutable internals. If we can't make it immutable, we should return a copy